analyzers/src/SonarAnalyzer.CSharp/Rules/PropertyToAutoProperty.cs

```diff
@@ -35,158 +35,123 @@ namespace SonarAnalyzer.Rules.CSharp
+                   && semanticModel.GetSymbolInfo(assignment.Right).Symbol is IParameterSymbol {Name: "value", IsImplicitlyDeclared: true}
+                   ? FieldSymbol(assignment.Left, semanticModel.GetDeclaredSymbol(setter).ContainingType, semanticModel)
+                   : null;
 
-            if (assignment != null &&
-                assignment.IsKind(SyntaxKind.SimpleAssignmentExpression) &&
-                assignment.Right != null &&
-                semanticModel.GetSymbolInfo(assignment.Right).Symbol is IParameterSymbol parameter &&
-                parameter.Name == "value" &&
-                parameter.IsImplicitlyDeclared)
-            {
-                return TryGetField(assignment.Left, semanticModel.GetDeclaredSymbol(setter).ContainingType,
-                    semanticModel, out setterField);
-            }
-            return false;
-
-            AssignmentExpressionSyntax GetAssignmentFromBody(BlockSyntax body) =>
-                body?.Statements.Count == 1 &&
-                body.Statements[0] is ExpressionStatementSyntax statement
+            AssignmentExpressionSyntax AssignmentFromBody(BlockSyntax body) =>
+                body?.Statements.Count == 1 && body.Statements[0] is ExpressionStatementSyntax statement
                 ? statement.Expression as AssignmentExpressionSyntax
                 : null;
 
-            AssignmentExpressionSyntax GetAssignmentFromExpressionBody(ArrowExpressionClauseSyntax expressionBody) =>
+            AssignmentExpressionSyntax AssignmentFromExpressionBody(ArrowExpressionClauseSyntax expressionBody) =>
                 expressionBody?.ChildNodes().Count() == 1
                 ? expressionBody.ChildNodes().ElementAt(0) as AssignmentExpressionSyntax
                 : null;
         }
 
-        private static bool TryGetField(ExpressionSyntax expression, INamedTypeSymbol declaringType,
-            SemanticModel semanticModel, out IFieldSymbol field)
+        private static IFieldSymbol FieldSymbol(ExpressionSyntax expression, INamedTypeSymbol declaringType, SemanticModel semanticModel)
         {
             if (expression is IdentifierNameSyntax)
             {
-                field = semanticModel.GetSymbolInfo(expression).Symbol as IFieldSymbol;
-                return field != null;
+                return semanticModel.GetSymbolInfo(expression).Symbol as IFieldSymbol;
             }
 
-            if (!(expression is MemberAccessExpressionSyntax memberAccess) ||
-                !memberAccess.IsKind(SyntaxKind.SimpleMemberAccessExpression))
+            if (!(expression is MemberAccessExpressionSyntax memberAccess)
+                || !memberAccess.IsKind(SyntaxKind.SimpleMemberAccessExpression))
             {
-                field = null;
-                return false;
+                return null;
             }
-
-            if (memberAccess.Expression is ThisExpressionSyntax)
+            else if (memberAccess.Expression is ThisExpressionSyntax)
```

Comment:
After this deobfuscation, it's now clear that the whole method is just wrong :D It should be like this:
```
            if (expression is IdentifierNameSyntax)
            {
                return semanticModel.GetSymbolInfo(expression).Symbol as IFieldSymbol;
            }
            else if (expression is MemberAccessExpressionSyntax memberAccess && memberAccess.IsKind(SyntaxKind.SimpleMemberAccessExpression))
            {
                return memberAccess.Expression is ThisExpressionSyntax
                    || (memberAccess.Expression is IdentifierNameSyntax identifier && semanticModel.GetSymbolInfo(identifier).Symbol is INamedTypeSymbol type && type.Equals(declaringType))
                    ? semanticModel.GetSymbolInfo(expression).Symbol as IFieldSymbol
                    : null;
            }
            else
            {
                return null;
            }
```